Service/Endpoint/EndpointFactory.php

```diff
@@ -144,6 +146,35 @@ private function setEndpointResponseClass(Endpoint $endpoint, $responseClass = n
         return $endpoint->setResponseClass($responseClass);
     }
 
+    /**
+     * @param Endpoint    $endpoint
+     * @param string|null $errorClass
+     *
+     * @return EndpointInterface
+     *
+     * @throws ConfigurationException
+     */
+    private function setEndpointErrorClass(Endpoint $endpoint, $errorClass = null): EndpointInterface
+    {
+        if (null === $errorClass) {
+            return $endpoint;
```

Comment:
Wouldn't it be easier to always have an `$errorClass` ? So that a default would be defined as well for generic purposes/existing implementations?

Reply:
Error class like other classes are in the ServiceAPIClientBundle repository. I'm worried about circular dependency.

Reply:
Error class like other classes are in the ServiceAPIClientBundle repository. I'm worried about circular dependency. Let's keep it that way